### PR TITLE
[feature][Doc] Add doc for `CommandAckResponse`

### PR DESCRIPTION
**Pull request by liangyepianzhou**
### Motivation
Now, the website do not have related doc of `CommandAckResponse` in https://github.com/apache/pulsar/pull/8996.
### Modification
Add the doc for `CommandAckResponse`.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `no-need-doc` 
(Please explain why)
  
- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-added`
(Docs have been already added)

### CODE REVIEW

File: site2/docs/developing-binary-protocol.md [hunk 1]
```diff
@@ -418,7 +418,24 @@ Parameters:
...
+If the `error` field is set, it indicates that the request has failed.
+
+Example of ack response with redirection:
+
+```protobuf
+message CommandAckResponse {
+    "consumer_id" : 1,
+    "txnid_least_bits" = 0,
+    "txnid_most_bits" = 1,
+    request_id = 5
```
Comment (RobertIndie) [inline site2/docs/developing-binary-protocol.md:436]:
```suggestion
    “request_id” = 5
```

File: site2/docs/developing-binary-protocol.md [hunk 2]
```diff
@@ -418,7 +418,24 @@ Parameters:
  * `txnid_least_bits` -> *(optional)* The ID of the transaction opened in a TC,
    `txnid_most_bits` and `txnid_least_bits`uniquely identify a transaction.
  * `request_id` -> *(optional)* ID for handling response and timeout.
-   
+
+##### Command CommandAckResponse
+
+This is the broker's response to ack request by the client. It contains the `consumer_id` sent in the request. 
+If transaction is used, it will also contain the transactionId and requestID sent in the request.
```
Comment (momo-jun) [inline site2/docs/developing-binary-protocol.md:425]:
```suggestion
If a transaction is used, it contains both Transaction ID and Request ID that are sent in the request.
```

File: site2/docs/developing-binary-protocol.md [hunk 3]
```diff
@@ -418,7 +418,24 @@ Parameters:
...
+The client will process the specific request according to the request ID.
```
Comment (momo-jun) [inline site2/docs/developing-binary-protocol.md:426]:
```suggestion
The client processes the specific request according to the Request ID.
```

File: site2/docs/developing-binary-protocol.md [hunk 4]
```diff
@@ -418,7 +418,24 @@ Parameters:
  * `txnid_least_bits` -> *(optional)* The ID of the transaction opened in a TC,
    `txnid_most_bits` and `txnid_least_bits`uniquely identify a transaction.
  * `request_id` -> *(optional)* ID for handling response and timeout.
-   
+
+##### Command CommandAckResponse
```
Comment (momo-jun) [inline site2/docs/developing-binary-protocol.md:422]:
```suggestion
##### Command AckResponse
```

File: site2/docs/developing-binary-protocol.md [hunk 5]
```diff
@@ -418,7 +418,24 @@ Parameters:
...
+
+This is the broker's response to ack request by the client. It contains the `consumer_id` sent in the request.
```
Comment (momo-jun) [inline site2/docs/developing-binary-protocol.md:424]:
```suggestion
An `AckResponse` is the broker’s response to acknowledge a request sent by the client. It contains the `consumer_id` sent in the request. 
```

**Review by momo-jun — APPROVED** [review]

**Comment by momo-jun** [comment]
Two more suggestions:
- Link to the code PR #8996.
- Add "[feature]" ahead of "[doc]" in the PR title.

**Comment by liangyepianzhou** [comment]
> Two more suggestions:
> 
> * Link to the code PR [Ack response implementation  #8996](https://github.com/apache/pulsar/pull/8996).
> * Add "[feature]" ahead of "[doc]" in the PR title.

In fact, the ACK Response has existed for a long time, and this PR just adds a request ID. This is also the reason why I didn't add this PR link before.

**Comment by momo-jun** [comment]
Does this doc change apply to version 2.8 or later? If yes, more historical versions of docs need update (site2/website/versioned_docs).

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**